assignment_2.rb

```diff
@@ -0,0 +1,68 @@
+class TreeNode 
+  attr_accessor :val,:l,:r 
+
+    def initialize(val, l=nil , r=nil)
+      @val = val
+      @l = l 
+      @r = r
+    end 
+end
+  
+# max = root node 
+def traverse(tree, max)
+  sum = 0
+  #check if val is higher than root
+  if tree.val >= max 
+    sum = 1 # if tree is valid, always count root 
+    sum += traverse(tree.l,max) if tree.l != nil 
+    sum += traverse(tree.r,max) if tree.r != nil 
+  else
+    # if val lower than root check left and right node 
+    sum += traverse(tree.l, max) if tree.l != nil 
+    sum += traverse(tree.r, max) if tree.r != nil
+  end 
+  sum
+end 
+
+def solution(tree)
+  return 0 if tree.val == nil
```

Comment:
May use this instead
```suggestion
  return 0 if tree.val.nil?
```